internal/controller/promotion/argocd.go

```diff
@@ -608,17 +649,21 @@ func (a *argoCDMechanism) applyArgoCDSourceUpdate(
 	source argocd.ApplicationSource,
 	newFreight []kargoapi.FreightReference,
 ) (argocd.ApplicationSource, error) {
-	if source.Chart != "" || update.Chart != "" {
+	logger := logging.LoggerFromContext(ctx)
+	updateLogger := logger.WithValues("source", source, "update", update)
+	updateLogger.Debug("About to apply ArgoCD source update")
 
-		if source.RepoURL != update.RepoURL || source.Chart != update.Chart {
-			// There's no change to make in this case.
+	if source.Chart != "" || update.Chart != "" {
+		// Kargo uses the "oci://" prefix, but Argo CD does not.
+		if source.RepoURL != strings.TrimPrefix(update.RepoURL, "oci://") || source.Chart != update.Chart {
+			updateLogger.Trace("Source update not applicable to Application source, skipping.")
```

Comment:
This was deliberately not done here before.

For simplicity, the "rule" for the updates is to perfectly match the thing you're applying the update to.

This means that for an app source update, you should match the source. This means you should _not_ use the oci:// prefix in your update.

(For a Helm dependency update, you should match the dependency. This means you _should_ use the oci:// prefix in your update.)

So the only time we really need to account for differences in using/not using oci is when comparing sources to _Freight_.

If there's any feeling that this should be re-evaluated, this isn't the PR to introduce the change.